src/org/traccar/database/DataManager.java

```diff
@@ -501,224 +268,52 @@ public void addStatistics(Statistics statistics) throws SQLException {
-                .setObject(driver)
-                .executeUpdate();
-    }
-
-    public void removeDriver(long driverId) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.deleteDriver"))
-                .setLong("id", driverId)
-                .executeUpdate();
-    }
-
-    public Collection<DriverPermission> getDriverPermissions() throws SQLException {
-        return QueryBuilder.create(dataSource, getQuery("database.selectDriverPermissions"))
-                .executeQuery(DriverPermission.class);
-    }
-
-    public void linkDriver(long userId, long driverId) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.linkDriver"))
-                .setLong("userId", userId)
-                .setLong("driverId", driverId)
-                .executeUpdate();
-    }
-
-    public void unlinkDriver(long userId, long driverId) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.unlinkDriver"))
-                .setLong("userId", userId)
-                .setLong("driverId", driverId)
-                .executeUpdate();
-    }
-
-    public Collection<GroupDriver> getGroupDrivers() throws SQLException {
-        return QueryBuilder.create(dataSource, getQuery("database.selectGroupDrivers"))
-                .executeQuery(GroupDriver.class);
-    }
-
-    public void linkGroupDriver(long groupId, long driverId) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.linkGroupDriver"))
-                .setLong("groupId", groupId)
-                .setLong("driverId", driverId)
-                .executeUpdate();
-    }
-
-    public void unlinkGroupDriver(long groupId, long driverId) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.unlinkGroupDriver"))
-                .setLong("groupId", groupId)
-                .setLong("driverId", driverId)
-                .executeUpdate();
-    }
-
-    public Collection<DeviceDriver> getDeviceDrivers() throws SQLException {
-        return QueryBuilder.create(dataSource, getQuery("database.selectDeviceDrivers"))
-                .executeQuery(DeviceDriver.class);
-    }
-
-    public void linkDeviceDriver(long deviceId, long driverId) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.linkDeviceDriver"))
-                .setLong("deviceId", deviceId)
-                .setLong("driverId", driverId)
-                .executeUpdate();
-    }
-
-    public void unlinkDeviceDriver(long deviceId, long driverId) throws SQLException {
-        QueryBuilder.create(dataSource, getQuery("database.unlinkDeviceDriver"))
-                .setLong("deviceId", deviceId)
-                .setLong("driverId", driverId)
+    public void removeObject(Class<? extends BaseModel> clazz, long entityId) throws SQLException {
+        String query = "database.delete" + clazz.getSimpleName();
```

Comment:
Of course it's possible to use reflection to construct name and get class by name, but I'm not sure it's a good idea. Maybe it's OK because we only get class and don't do any iterations.